### PR TITLE
Fix: Add `skip-link` ID to `Skip to content` anchor tag

### PR DESCRIPTION
**Pull request by Debarghya-Banerjee**
Trac Ticket: Core-62311

## Summary

- This pull request introduces an ID attribute to the "skip to content" link, enhancing accessibility for users navigating with assistive technologies. By adding `id="skip-link"`, we ensure that this link can be targeted by "back to top" functionality, providing a seamless user experience.

## Changes Made

- Added `id="skip-link"` to the existing `"skip to content"` link.

## Rationale

- While the current implementation allows users to visually jump to the content, adding an ID ensures that:

- The "skip to content" link can be easily referenced by other navigation elements, such as a "back to top" link.

- When users activate these links, they receive proper focus, improving the overall accessibility of the site.

- This small change can significantly enhance the experience for keyboard and screen reader users, making navigation more intuitive.

## Impact

- This update will improve compliance with accessibility standards and create a better user experience for all visitors, particularly those using assistive technologies.

### CODE REVIEW

**Review by apermo — APPROVED** [review]
Looking good to me.

File: src/wp-includes/theme-templates.php [hunk 1]
```diff
@@ -203,6 +203,7 @@ function wp_enqueue_block_template_skip_link() {
 		// Create the skip link.
 		skipLink = document.createElement( 'a' );
 		skipLink.classList.add( 'skip-link', 'screen-reader-text' );
+		skipLink.id = 'skip-link';
```
Comment (sabernhardt) [inline src/wp-includes/theme-templates.php]:
```suggestion
		skipLink.id = 'wp-skip-link';
```

`wp-skip-link` might help avoid duplicate IDs

Comment (Debarghya-Banerjee) [reply src/wp-includes/theme-templates.php]: Fixed

**Comment by Debarghya-Banerjee** [comment]
Hi @joedolson, @sabernhardt 

I have updated the id to `wp-skip-link`.

Can you please check?
Thanks.

**Comment by philliproth** [comment]
Hey @apermo @sabernhardt - would love to see this in 6.8 - thank you for reviewing!

**Review by joedolson — APPROVED** [review]

**Comment by joedolson** [comment]
In r59559

**Comment by dilipom13** [comment]
Hello @Debarghya-Banerjee yes wp-skip-link is working properly.

**Pull request closed without merging**